devito/finite_differences/derivative.py

```diff
@@ -277,8 +285,25 @@ def evaluate(self):
         # types of discretizations.
         return self._eval_fd(self.expr)
 
+    @property
+    def _eval_deriv(self):
+        return self._eval_fd(self.expr)
+
     def _eval_fd(self, expr):
-        expr = getattr(expr, 'evaluate', expr)
+        """
```

Comment:
I like how the new docstring is coming along. Here's a suggestion. I'd remove
```
For example `f.dx * g` will evaluate `f.dx` first
```

at point 1, and I'd add a _complete_ example that shows the evolution of evaluation at each of the four steps. For example:

```
Evaluate the finite difference approximation of the Derivative.

The evaluation consists of four steps:
<the four points here>

Examples
--------
Consider:

f = Function(...)
...

After step 1:
...

After step 2:
...
```

I think we need such a thorough docstring because this method is too crucial and we need to make sure people (including me) understands exactly what's going on and why

Reply:
FWIW, I still think we need 

```
Examples
--------
...
```

Reply:
I think it's pretty clear as it is, it's an internal method